File: Python3/1201 - 1400/1299 Replace Elements with Greatest Element on Right Side.py
# 132ms, 27.45%; 14.1MB, 100.00%
class Solution:
    def replaceElements(self, arr: List[int]) -> List[int]:
        res = arr[:-1] + [-1]
        for i in range(len(arr) - 1, 0, -1):
            res[i - 1] = max(res[i], arr[i])
        return res

File: 1201 - 1400/1299. Replace Elements with Greatest Element on Right Side.cc
// 48ms, 94.53%; 11MB, 100.00%
#include <bits/stdc++.h>
using namespace std;

class Solution {
public:
    vector<int> replaceElements(const vector<int> &arr) {
        vector<int> res(arr.begin(), arr.end() - 1);
        res.push_back(-1);
        for (int i = arr.size() - 1; i > 0; --i) res[i - 1] = max(res[i], arr[i]);
        return res;
    }
};
